promgen/views.py

```diff
@@ -1019,13 +1019,28 @@ def post(self, request, *args, **kwargs):
         # when we run tasks.process_alert
         alert = models.Alert.objects.create(body=request.body.decode("utf-8"))
         tasks.process_alert.delay(alert.pk)
+        tasks.index_alert.delay(alert.pk)
         return HttpResponse("OK", status=202)
 
 
 class AlertList(LoginRequiredMixin, ListView):
     paginate_by = 20
-    queryset = models.Alert.objects.order_by('-created')
 
+    def get_context_data(self, **kwargs):
```

Comment:
I think I would probably leave the queryset at the top, and then implement it something like 

```python
class AlertList(LoginRequiredMixin, ListView):
    paginate_by = 20
    queryset = models.Alert.objects.order_by("-created")

    def get_queryset(self):
        search = self.request.GET.get("search")
        if search:
            return self.queryset.filter(
                Q(alertlabel__name="Service", alertlabel__value__icontains=search)
                | Q(alertlabel__name="Project", alertlabel__value__icontains=search)
            )
        return self.queryset
```

Reply:
fixed